Tickets/dm 15348

wmwv opened the pull request:
Defines CI, SMALL, MEDIUM, LARGE datasets.

wmwv committed:
Add discussion of dataset goals and specific examples.

wmwv committed:
Add simple checks under CI dataset.  Number of Stars, CCDs.

wmwv committed:
Note that HiTS is primarily g-band.

wmwv committed:
Add simple discussion of current validation_data_* usage.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+    * Steps
+        - ISR
+        - processCcd
+        - ~Coadd~
```
Why the `~`? Are these approximate coadds?

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+    * Goals
+        - Fuller integrated testing
+        - Verify DIA runs
+        - Allow checks for reasonable:
```
What's the difference between this test and "reasonable ranges of" in CI?

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+        - Verify that DIA works
+    * Goals
+        - Fuller integrated testing
+        - Verify DIA runs
```
This seems redundant with "Verify that DIA works". Is this a formal process thing?

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+    * Goals
+        - Test that key initial processing steps execute
+        - Allow checks for reasonable ranges of
+            - Fraction of CCDs successfully processed
```
I would have thought that CI datasets would be vetted for "known good" data, and that we can therefore expect all observations to be successfully processed. Is this something we want to deal with at the CI scale? You mention "robustness" under medium-scale datasets...

wmwv replied:
That's a good point.  Currently 1-2 CCDs of the `validation_data_decam` set are expected to fail.  But I agree that we should change that so that CI should always pass and either SMALL or MEDIUM catches edge cases.  I wrote this under MEDIUM so I'll keep it there, but I'm a bit ambivalent.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+Jenkins vs. NCSA
+================
+The above goals and dataset definitions are written with the NCSA Verification Cluster in mind.
+The current Jenkins AWS solution has a much smaller number of available cores than the NCSA Verification Cluster.  The limitations imposed by that mean that a more restricted set of minimal data will be necessary.  This more limited set of data may also be appropriate for use on an individual machine for direct developer testing.
```
How do the capabilities of Jenkins compare to what you're requiring for CI? (It might also help to explicitly say in the CI section that you don't mean Jenkins or Travis).

I wouldn't consider "15 minutes on 16 cores" an onerous requirement for a laptop run. In fact, the running time would be dominated by updating the Stack...

wmwv replied:
I believe Jenkins is currently 4-8 cores.  The `validation_data_hsc` daily runs are done on a special medium scale node.

The longer-term plan is that CI is done at NCSA, which will have more easily deployed >16-core nodes for CI.

And yes, you're right that the CI-scale, even under the NCSA full capabilities should be runnable by a developer on their individual machine.  I'll add that to Requirements and update the "Jenkins vs. NCSA" discussion.

And thanks for prompting me to think that I should also call out the build vs. test part to clarify that the time requirements are meant to be just the science pipelines part.

kfindeisen replied:
The new text still implies that the CI datasets should not be run on Jenkins. Was that the intent?

wmwv replied:
Yesish.  These are future-focused definitions.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+    * The HSC PDR that is currently processed bi-weekly satisfies needs for Large datasets
+    * Modulo
+      1. DIA
+      2. testing of ingestion of DRP data.
```
Not clear what you mean here. Are you saying that the data might not be useable?

Also, this doesn't format correctly in the compiled page. I think you need a blank line before "1."

wmwv replied:
I meant that the data weren't being currently ingested.  But I realize I'm mixing implementation and dataset definition.  I've removed the ingestion line.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+4. CI
+    a. HSC Engineering data "ci_hsc"
+      - 8 GB of data.  Runs through single-frame, coadd, and forced photometry.
+      - Takes several hours when running on only a few nodes.
```
If it takes several hours despite access to multiple nodes (I assume this means at least a dozen cores), is it actually CI-sized by your definition above?

wmwv replied:
You're right, despite the name, this is clearly SMALL in the definition above, and in use.  It's not part of the basic CI, but it is used by developers to check that things still work when going all the way through to coaddition.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+      - Takes several hours when running on only a few nodes.
+    b. DECam HiTS
+      - A subset of data intended for CI AP testing (with Blind15A_40 and Blind15A_42) is in
+        https://github.com/lsst/ap_verify_ci_hits2015
```
It's not just specific fields: the data were also shrunk by using only 3 visits and 2 ccds per visit.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+        Presently (2018-08-15) the data are on a branch, not yet merged to master.
+    c. validation_data_decam, validation_data_cfht
+      - Each of these is part of CI and regularly used for simple execution testing.
+      - ISR is not performed.
```
Might want to say what is performed; as far as I know DIA isn't exercised either.

wmwv replied:
DIA isn't part of the requirements for a CI-scale dataset.  I've added a note to the list to explicitly load this into the reader's mind again.

kfindeisen replied:
Ah, I didn't understand that you were using `~` to mean "not" in the earlier version. Objection withdrawn.

kfindeisen commented on index.rst:
```diff
@@ -1,40 +1,148 @@
+      - Each of these is part of CI and regularly used for simple execution testing.
+      - ISR is not performed.
 
 :tocdepth: 1
```
This got corrupted somehow; maybe it needs to be at the start of the file rather than the end?

Likewise, the "This technote is not yet published" warning is not very useful at the end of the document. 😉

wmwv replied:
Yes, thank you.  I've reread through the template in `index.rst` and re-ordered things hopefully more correctly.

kfindeisen reviewed (changes requested):
I found the document a bit hard to understand because it reads like a set of notes rather than a report. In addition to the specific lines I've commented, I found the goals for the different size classes a bit vague (e.g., "integrated testing" vs. "verification" vs. "quantitative performance"). Could you expand to a few sentences about what is and isn't being covered at each level?

wmwv committed:
Note that master calibration generation will be part of future work.

wmwv committed:
Remove strikethrough markup which doesn't work in Sphinx+rst.

wmwv commented:
Thanks for the comments, @kfindeisen.

It is indeed more of a set of internal notes rather than a report.  I talked things over with @SimonKrughoff and we'll incrementally merge to master to circulate among a small team, but it's certainly true that this won't be meant for wider distribution until it has more narrative and context.

I'll go through and take your comments to improve the current version, and then continue working on future branches to bring this document closer to converging to something more widely useful.

wmwv committed:
Move tocdepth adn related to start of file.

wmwv committed:
Move sections up a level.

wmwv committed:
Change name to Scientific Performance Monitoring.

wmwv committed:
Clarify requirements goals of each data set type.

Updating in response to useful comments by Findeisen.

wmwv committed:
Make all list indents 4 spaces.

wmwv committed:
Move ci_hsc -> SMALL example dataset.

Remove note about the HSC PDR processing not being suitable for DRP ingest.
That was incorrect.  It's suitable, it's just not being done.

wmwv commented:
To complete the Jenkins answer above: I believe the current `ci_hsc` and `validation_data_hsc` Jenkins runs are using AWS `m4.xlarge`:  4 cores, 16 GB.

Running directly on even just a single NCSA node can be significantly faster.
